app/models/manageiq/providers/vmware/infra_manager/inventory/persister.rb

```diff
@@ -51,7 +51,7 @@ def initialize_inventory_collections
     add_collection(infra, :storage_profiles)
     add_collection(infra, :storage_profile_storages)
     add_collection(infra, :parent_blue_folders)
-    add_collection(infra, :vms_and_templates) do |builder|
+    add_collection(infra, :vms_and_templates, {}, {:without_sti => true}) do |builder|
```

Comment:
Since vms and templates are in the same collection we can't use auto-model-class here, the parser already sets `:type =>` explicitly